theano/tensor/nnet/conv.py

```diff
@@ -34,6 +34,87 @@
 _logger = logging.getLogger("theano.tensor.nnet.conv")
 
 
+def get_conv2d_output_shape(image_shape, kernel_shape,
+                            border_mode, subsample):
+    """
+    This function compute the output shape of convolution operation.
+
+    Parameters
+    ----------
+    image_shape: tuple of int corresponding to the input image shape. Its four
+        element must correspond respectively to: batch size, number of input
+        channels, height and width of the image. None where undefined.
+    kernel_shape: tuple of int corresponding to the kernel shape. Its four
+        elements must correspond respectively to: number of output channels,
+        number of input channels, height and width of the kernel. None
+        where undefined.
+    border_mode: string or tuple of int. If it is a string, it must be
+        'valid', 'half' or 'full'. If it is a tuple, its two elements
+        respectively correspond to the padding on height and width axis.
+    subsample: tuple of int. Its two elements espectively correspond to the
+        subsampling on height and width axis.
+
+    Returns
+    -------
+    output_shape: tuple of int corresponding to the output image shape. Its
+        four element must correspond respectively to: batch size, number of
+        output channels, height and width of the image. None where undefined.
```

Comment:
Actually, these functions are supposed to work with Theano scalar variables of integer dtype, not only Python ints. In fact, when they are called through `infer_shape`, their inputs will be symbolic scalars.